packages/vue-app/template/components/nuxt-link.client.js

```diff
@@ -0,0 +1,89 @@
+<%= isTest ? '// @vue/component' : '' %>
+import Vue from 'vue'
+
+const requestIdleCallback = window.requestIdleCallback ||
+  function (cb) {
+    const start = Date.now()
+    return setTimeout(function () {
+      cb({
+        didTimeout: false,
+        timeRemaining: function () {
+          return Math.max(0, 50 - (Date.now() - start))
+        },
+      })
+    }, 1)
+  }
+const observer = window.IntersectionObserver && new window.IntersectionObserver(entries => {
+  entries.forEach(({ isIntersecting, target: link }) => {
+    if (!isIntersecting) {
```

Comment:
This could be changed to use `intersectionRatio` instead, for maximum browser support (https://github.com/w3c/IntersectionObserver/issues/211).

```js
if (intersectionRatio <= 0) {
```

Reply:
Updated, thanks @maoberlehner :)